pkg/kubectl/cmd/completion.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/kubectl/pkg/cmd/util"
+)
+
+type filterFunc func(config clientcmdapi.Config, toComplete string) []string
+
+func getFromConfig(f util.Factory, filter filterFunc) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		config, err := f.ToRawKubeConfigLoader().RawConfig()
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		return filter(config, toComplete), cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func registerCompletionFuncForGlobalFlags(cmds *cobra.Command, f util.Factory) error {
```

Comment:
Just a nit. Why `cmds` with an 's'?  It's just a single command, no? I'm not sure why the calling function uses `cmds` with an 's'.

Reply:
just copy-paste from the original code, we could trim the `s`.